goalie/go_mesh_seq.py

```diff
@@ -61,8 +62,12 @@ def get_enriched_mesh_seq(
             get_qoi=self._get_qoi,
             get_bcs=self._get_bcs,
             qoi_type=self.qoi_type,
+            parameters=self.params,
         )
 
+        # update self._fs variable for enriched mesh
+        mesh_seq_e.function_spaces
```

Comment:
Please could you introduce a method on the `MeshSeq` to do this whose name says what it is doing, e.g., `update_function_spaces`? On it's own, this line of code looks like it's doing nothing. (My bad for writing it this way in the first place, I was probably overusing properties.)

Reply:
@jwallwork23 
I added a function called `update_function_spaces` which calls `function_spaces` if the function space attribute dictionary is `None`. Hopefully, that is a bit more readable.